File: backend/app/core/upgrades/curses/shattered_bullet_1.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class ShatteredBullet1(UpgradeImplementation):
    id = 'shattered_bullet_1'
    name = 'Shattered Bullet 1'
    description = '-1 bullet damage.'
    rarity = 'uncommon'
    type = 'stat_modifier'
    cost = 0

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = -1
        appliers.modifiers.add_multiplier('bullet', 'damage', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True

File: frontend/src/game/upgrades/curses/shattered_bullet_1.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const ShatteredBullet1Def: UpgradeDef = {
  id: "shattered_bullet_1",
  name: "Shattered Bullet 1",
  description: "-1 bullet damage.",
  rarity: RarityID.Uncommon,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 0,
  targetClass: UpgradeTargetID.Bullet,
  fieldInTargetClass: UpgradeStatID.Damage,
  value: -1,
  isMultiplier: false,
  stackable: true,
  maxStacks: 99999,
  curse: true,
}

export class ShatteredBullet1 extends Upgrade {}
